File: openflix-frontend/src/app/core/models/favorites/favorite.ts
import { Serie } from "../videos/serie";

export class Favorite {
    public series: Array<Serie>

    constructor(series: Array<Serie> = []) {
        this.series = series
    }

    checkIfIsFavorite(serie: Serie): boolean {
        const index = this.series.findIndex(findSerie => findSerie.id === serie.id)
        return index >= 0
    }
}

File: openflix-api/OpenFlixAPI/OpenFlixAPI/Domain/Models/Videos/Favorite.cs
using OpenFlixAPI.Domain.Models.Profiles;
using OpenFlixAPI.Domain.Models.Users;

namespace OpenFlixAPI.Domain.Models.Videos
{
    public class Favorite
    {
        public int Id { get; set; }

        public int ProfileId { get; set; }
        public Profile Profile { get; set; }

        public int UserId { get; set; }
        public User User { get; set; }

        public IEnumerable<Serie> Series { get; set; }

        public override string ToString()
        {
            return $"Favorite(Id: {Id}, ProfileId: {ProfileId}, Profile: {Profile}, UserId: {UserId}, User: {User}, Series: {Series})";
        }
    }
}
